plotly/session.py

```diff
@@ -115,6 +117,13 @@ def update_session_plot_options(**kwargs):
             raise exceptions.PlotlyError("{} must be of type '{}'"
                                          .format(key, PLOT_OPTIONS[key]))
 
+        # raise exception if sharing is invalid
+        if (key == 'sharing' and not (kwargs[key] in
+                                      ['public', 'private', 'secret'])):
+            raise exceptions.PlotlyError("'{}' must be of either {}"
+                                         .format(key, "'public', 'private'" +
+                                                 " or 'secret'"))
```

Comment:
:dromedary_camel: (DRY :cow2:) You're repeating `'public', 'private', 'secret'`

There should be a variable somewhere that controls that, like `SHARING_OPTIONS`. then you can use it in these two places and it can remain centralized in the future.